torch/_inductor/lowering.py

```diff
@@ -5398,6 +5398,9 @@ def fill_(x, fill_value):
 
 @register_lowering(aten.copy_, type_promotion_kind=None)
 def copy_(dst, src, non_blocking=False):
+    if dst is src:
+        # dst.copy_(dst) can happen from the reinplacing pass
+        return dst
```

Comment:
The alternative is to change the reinplacing pass to handle this. That is very difficult due to the current design. The reinplacing pass doesn't actually remove auto_functionalized and triton_kernel_wrapper_functional operators from the graph: instead, during their lowering phase, we replace those nodes with clones and a mutable operation.

Reply:
@eellison are you suggestion a separate pass that decomposes the wrappers and then another separate pass to remove x.copy_(x)?

Handling the removal of `x.copy_(x)` in the lowering for copy_ seemed less complicated

Reply:
Depends on if we have other transformations we want to do. i guess this is fine as a peephole but generally better to do graph transformations in fx layer..